quinn/src/connection.rs

```diff
@@ -799,17 +798,31 @@ impl<S> ConnectionInner<S>
 where
     S: proto::crypto::Session,
 {
-    fn drive_transmit(&mut self) {
+    fn drive_transmit(&mut self) -> bool {
         let now = Instant::now();
+        let mut transmits = 0;
 
         let max_datagrams = caps().max_gso_segments;
 
         while let Some(t) = self.inner.poll_transmit(now, max_datagrams) {
+            transmits += match t.segment_size {
+                None => 1,
+                Some(s) => (t.contents.len() + s - 1) / s, // round up
+            };
             // If the endpoint driver is gone, noop.
             let _ = self
                 .endpoint_events
                 .unbounded_send((self.handle, EndpointEvent::Transmit(t)));
+
+            if transmits >= MAX_TRANSMIT_DATAGRAMS {
+                // TODO: What isn't ideal here yet is that we don't poll all
```

Comment:
Missing "if" in "is that we don't"? Maybe cite #1126?

Reply:
linked